src/Catalog.Infrastructure/DataAccess/Repositories/CategoryRepository.cs

```diff
@@ -36,15 +37,37 @@ OUTPUT INSERTED.Id
                 ParentId = category.Parent?.Id
             };
 
-            return await connection.QuerySingleAsync<int>(sql, parameters);
+            try
+            {
+                return await connection.QuerySingleAsync<int>(sql, parameters);
+            }
+            catch (SqlException ex)
+            {
+                if (ex.Number == 547)
+                    throw new CategoryConflictException($"Parent category with id: {category.Parent?.Id} does not exist");
+
+                throw;
+            }
         }
 
         public async Task Delete(int id)
         {
-            var sql = @"DELETE FROM Category WHERE Id = @Id";
+            var sql = @"DELETE FROM Item WHERE CategoryId = @Id
+                    DELETE FROM Category WHERE Id = @Id";
```

Comment:
How deletion of related items is implemented? `Delete category (with the related items)`

Reply:
The SQL deletes the items which has the provided Category Id, and then deletes the category as well.